org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -190,8 +195,37 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 				} else if (source instanceof List) { // ambiguous
 					@SuppressWarnings("unchecked")
 					List<Object> matches = (List<Object>) source;
+					List<Object> exactMatchesFiltered = new ArrayList<>();
+					String originalHyperLink2 = originalHyperLink;
+					int firstMethodStartIndex = originalHyperLink2.indexOf("."); //$NON-NLS-1$
+					int firstMethodClosing = originalHyperLink2.indexOf(")"); //$NON-NLS-1$
+					if (firstMethodStartIndex != -1 && firstMethodClosing != -1) {
+						String firstMethod = originalHyperLink2.substring(firstMethodStartIndex + 1, firstMethodClosing + 1);
+						for (Object obj : matches) {
+							@SuppressWarnings("restriction")
+							SourceType source = (SourceType) obj;
+							try {
+								@SuppressWarnings("restriction")
+								IMethod[] methods = source.getMethods();
+								for (IMethod method : methods) {
+									int indexOfClosing = method.toString().indexOf(")"); //$NON-NLS-1$
+									int indexOfStart = method.toString().substring(0,indexOfClosing+1).lastIndexOf(" "); //$NON-NLS-1$
```

Comment:
`lastIndexOf(" ")` may work in this situation but generally you should avoid matching for space, as other or multiple whitespaces would be legal too. So search for something meaningfull

Reply:
In this specific case multiple whitespaces are occurred prior to the actual function name, here the toString() always returns values in this format -> _"**static void test() [in Sample [in Sample.java [in <default> [in src [in New]]]]]**"_
and for every extracted methods there was only one closing and one opening, which is part of the actual function name, so my approach was put the closing brace as the limit of the substring and access the last whitespace which always results in getting the whitespace prior to the actual function name, there weren't any direct methods for extracting method name & signature only